### PR TITLE
Add events for image uploads

### LINKED ISSUE #84: Throw events when an image has been uploaded
Hello,

I currently have an app where the uploaded images need to be converted to the webp format. Currently I do this by observing my model, looping over the editor blocks and sending out jobs to convert the images from the block data. However, it would be nice if this package could thrown an event whenever an image has been successfully uploaded.

What do you think about this idea? I can implement it I think. Just wanted to get an opinion of the maintainer(s) first before I sink some time in it.

It would definitely make our codebase cleaner, and I'm sure other people can find a use for the events if they need to do some extra post-processing with images.

Cheers.

### PR DESCRIPTION
**Pull request by Woeler**
This PR adds events for image uploads and thumbnail generation. This allows users to hook into these processes. I for instance want to convert uploaded images to webp, and use the OG image as fallback. With these events I can now just listen for uploads and dispatch conversion jobs.

I'm sure there are many other things people will be able to do with this feature.

Resolves #84

### CODE REVIEW

**Review by roelofr — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,5 @@
 .DS_Store
 Thumbs.db
 .php-cs-fixer.cache
+/auth.json
+/.ddev/
```
Comment (roelofr) [inline .gitignore:9]: What are these files for, they seem unused.

If it's something you need to ignore locally, please add them to either a [global ignore file][1] or in the `.git/info/exclude` file in your repository.

[1]: https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files#configuring-ignored-files-for-all-repositories-on-your-computer

**Comment by Woeler** [comment]
Thanks for your comments. I will update this pr this week!

**Comment by Woeler** [comment]
I am unsure why the v8 tests keep failing. Running them locally produces no such errors.

**Comment by Woeler** [comment]
@roelofr Alright, everything seems to be good now

**Review by roelofr — CHANGES_REQUESTED** [review]
Hey, thanks for picking this back up after the pipelines were fixed.

As time moves on, so does my understanding of code, It seems like a good idea to remove unused traits, but other than that it looks ready to roll.

Please add a CHANGELOG entry so we can keep track of these changes, then we'll get this merged in.

**Review by roelofr — APPROVED** [review]

**Pull request merged**